lib/concurrent/atomic/atomic_fixnum.rb

```diff
@@ -74,10 +68,7 @@ def value=(value)
     #
     #   @return [Fixnum] the current value after incrementation
     def increment
-      @mutex.lock
-      @value += 1
-    ensure
-      @mutex.unlock
+      synchronize { @value += 1 }
```

Comment:
Yeah I understand your concern, it should be probably mentioned in the documentation that it does not convert, there will always be some space for misunderstanding. Still this is the best naming we have. I guess this will be used by advanced users only so I am not worried that much. It's primarily an Atomic class with suffix describing what type does it supports. There would have to be AtomicBignum to suggest some conversion, but even then it is not compatible with what atomic does - a reference, it cannot auto-convert itself.